src/OrchardCore.Modules/OrchardCore.Tenants/Views/Admin/Index.cshtml

```diff
@@ -110,61 +77,20 @@
-                                            <span class="badge ta-badge font-weight-normal"><i class="fa fa-hourglass-start text-info" aria-hidden="true"></i> @T["Initializing"]</span>
-                                            break;
-                                        case TenantState.Running:
-                                            <span class="badge ta-badge font-weight-normal"><i class="fa fa-check text-success" aria-hidden="true"></i> @T["Running"]</span>
-                                            break;
-                                        case TenantState.Disabled:
-                                            <span class="badge ta-badge font-weight-normal"><i class="fa fa-remove text-danger" aria-hidden="true"></i> @T["Disabled"]</span>
-                                            break;
-                                        case TenantState.Invalid:
-                                            <span class="badge ta-badge font-weight-normal"><i class="fa fa-exclamation-triangle text-warning" aria-hidden="true"></i> @T["Invalid"]</span>
-                                            break;
-                                    }
-                                </a>
+    
+                                @await DisplayAsync(tagsShape)
                             </div>
                         </div>
                         <div class="col-lg-6 col-sm-12 float-end">
                             <div class="float-end">
-                                <a asp-action="Edit" asp-route-id="@entry.Name" class="btn btn-primary btn-sm edit">@T["Edit"]</a>
-                                @if (entry.ShellSettings.State == TenantState.Running)
-                                {
-                                    <a class="btn btn-info btn-sm view" href="@GetEncodedUrl(entry, originalPathBase, appendQuery:false)" target="_blank">@T["View"]</a>
-                                }
-                                @if (!entry.IsDefaultTenant && entry.ShellSettings.State == TenantState.Running)
-                                {
-                                    <a asp-action="Disable" asp-route-id="@entry.Name" class="btn btn-danger btn-sm disable" data-url-af="RemoveUrl UnsafeUrl" data-title="@T["Disable"]" data-message="@T["Are you sure you want to disable this tenant?"]" data-ok-text="@T["Ok"]" data-cancel-text="@T["Cancel"]">@T["Disable"]</a>
-                                }
-                                @if (entry.ShellSettings.State == TenantState.Disabled)
-                                {
-                                    <a asp-action="Enable" asp-route-id="@entry.Name" class="btn btn-success btn-sm enable" data-url-af="UnsafeUrl">@T["Enable"]</a>
+                                @{
+                                    var buttonShape = await DisplayManager.BuildDisplayAsync(entry, UpdateModelAccessor.ModelUpdater, String.Empty, "ActionButtons");
```

Comment:
The controller is an updated so it doesn't have to be injected. Or better, move the calls to the display manager in the controller, and pass the shapes to the view model.

Reply:
@sebastienros I changed this by creating a shape instead of using Display Drivers. Let me know if you see a better way to doing this